### ISSUE TITLE
Update the official tutorial, "Groundwork" part

### ISSUE THREAD

**Issue by NicklGames** [HaxeFlixel/flixel#2523]
### Discussed in https://github.com/HaxeFlixel/flixel/discussions/2521

<div type='discussions-op-text'>

<sup>Originally posted by **NicklGames** March 16, 2022</sup>
Let me explain myself

If following the tutorial, when reaching this part
![image](https://user-images.githubusercontent.com/60806448/158607234-2ca3afbe-3de7-462f-874c-7c9051676695.png)
this will cause a bug, in which player will constantly move because it's telling to move the player every frame with no condition

It can be easily fixed tho, in 2 ways i know
1: Setting the speed to something only when player is moving and setting it to 0 when player is not
![image](https://user-images.githubusercontent.com/60806448/158607675-9f708e01-b8a1-47ec-9d82-34f830a2e80a.png)
![image](https://user-images.githubusercontent.com/60806448/158607724-1ee4898b-ec28-4f20-9570-de881beb139d.png)

2: Adding a boolean that will tell if player is actually moving and move the player only if any of the keys is actually pressed
![image](https://user-images.githubusercontent.com/60806448/158608020-b5bc30c9-8bc3-4367-b203-5634ce502e93.png)
![image](https://user-images.githubusercontent.com/60806448/158608072-bb70b2cb-c8c8-47e0-9c97-437d97dafedd.png)

It really is easy, why not update the tutorial (if the reason that it IS easy, then why letting an intentional bug be in an OFFICIAL tutorial?)</div>

**Comment by NulllStack** [comment]
I don't think so, if you have checked the [project](https://github.com/HaxeFlixel/flixel-demos/blob/dev/Tutorials/TurnBasedRPG/source/Player.hx) the tutorial made, you'll find that the `Player.hx` uses the  same code the tutorial have shown.

**Comment by NulllStack** [comment]
the reason that cause the player moves in a certain direction constantly is probably that you forgot to set the `drag.x = drag.y = 1600` ?

**Comment by NulllStack** [comment]
and there's another thing, you'll have to put `velocity.set(SPEED,0)` and `velocity.rotate(FlxPoint.weak(0,0), newAngle` inside the `if (up || down || left || right)` condition due to only execute it only if player pressed the move keys.

**Comment by NulllStack** [comment]
thus if the player not pressing any move keys, the velocity is affected by the `drag.x = drag.y = 1600` and it will go down to 0, then the character will stop moving

**Comment by Geokureli** [comment]
I'll add this to the 5.0.0 release. It might be time to revisit this tutorial from the ground up, since newcomers seem to have a lot of friction.

**Comment by NulllStack** [comment]
Hi! @Geokureli , I did checked the discussion #2521 and Im pretty sure the tutorial is correct, the thing is that reader may easily forgot to put `velocity.set()` and `velocity.rotate()` inside the `if()` body that described on 6th step .

**Comment by NulllStack** [comment]
However, this tutorial has been there for a long time, it would be great to have something new for the newcomer tho :D

**Comment by Geokureli** [comment]
I think the tutorial should be revisited, this part here is quite clunky and only serves to confuse the learner with how roundabout it is for such a basic function. but also yes, we should add new tutorials as well
